### ISSUE TITLE
chore: fix the mypy baseline of tests and examples and widen the ci type-check step

### ISSUE THREAD

**Issue by g8rdier** [g8rdier/hqnn-forge#145]
## Context

#42 adds a `lint` CI job whose type-check step runs `mypy hqnn_forge` only. The package is clean after #102, but the tests and the example are not:

```
$ mypy tests examples
Found 54 errors in 7 files (checked 11 source files)
```

By file: `tests/test_preprocessing.py` 37, `tests/test_batched_forward.py` 8, `examples/quick_start.py` 4, `tests/test_modes.py` 2, one each in `test_hybrid_classifier.py`, `test_integration.py`, `test_parallel_hybrid_classifier.py`.

By kind: 20 `union-attr`, 17 `arg-type`, 11 `operator`, 6 `index`. Three patterns account for nearly all of them:

- `PCANormalizer` fitted attributes are typed `ndarray | None`, so every test that reads `normalizer.components_[0]` or `len(normalizer.mean_)` after `fit` is an `arg-type` / `index` / `operator` error on the `None` half of the union (all 37 in `test_preprocessing.py`).
- `model.classical_encoder[0]` and `.register_forward_pre_hook` on `nn.Sequential` children: `nn.Module.__getattr__` returns `Tensor | Module`, so indexing and hook registration are `union-attr` / `index` errors.
- `examples/quick_start.py` mixes `Tensor` and `ndarray` arithmetic in the reporting code.

## Expected Outcome

- Decide per pattern: `assert normalizer.components_ is not None` after `fit` in tests (or a typed `fitted_` accessor on the class), `cast(nn.Sequential, ...)` / `typing.cast` for the module indexing, and explicit `.numpy()` / `.item()` in the example.
- `mypy hqnn_forge tests examples` exits 0.
- Widen the type-check step in `.github/workflows/tests.yml` from `mypy hqnn_forge` to all three trees.

## Notes

Found while implementing #42; deliberately left out of that PR so the CI job could start green from the existing #102 baseline instead of growing a typing pass across 7 files.

## Type

Chore
